src/components/taskDetails/Details.tsx

```diff
@@ -6,15 +6,65 @@ import extractRepoName from '@/utils/extractRepoName';
 import Link from 'next/link';
 import { FaReceipt } from 'react-icons/fa6';
 import DevFeature from '../DevFeature';
+import moment from 'moment';
+import Tooltip from '@/components/common/Tooltip/Tooltip';
 
 const Details: FC<TaskDetailsProps> = ({ detailType, value, url }) => {
     const color = value ? setColor?.[value] : undefined;
     const isGitHubLink = detailType === 'Link';
     const gitHubIssueLink = isGitHubLink ? value : undefined;
 
+    const getRelativeTime = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).fromNow() : 'N/A';
+    };
+
+    const getTooltipText = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).format('LLLL') : 'N/A';
+    };
+
+    const isPastDate = (timestamp: string | number | undefined) => {
+        return timestamp ? moment(timestamp).isBefore(moment()) : false;
+    };
+
+    const getHumanReadableTime = (timestamp: string | number | undefined) => {
+        if (!timestamp) return 'N/A';
+
+        const now = moment();
+        const futureTimestamp = moment(timestamp);
+
+        if (futureTimestamp.isAfter(now)) {
+            const yearsDiff = futureTimestamp.diff(now, 'years');
+            return `in ${yearsDiff} years`;
+        } else {
+            const duration = moment.duration(now.diff(futureTimestamp));
+            return `${duration.humanize()} ago`;
+        }
+    };
+
+    const isTimeDetail =
+        detailType === 'Started On' || detailType === 'Ends On';
+
+    const isAssigneeOrReporter =
+        detailType === 'Assignee' || detailType === 'Reporter';
+
+    const isTypeOrPriorityOrStatus =
+        detailType === 'Type' ||
+        detailType === 'Priority' ||
+        detailType === 'Status';
+
+    const renderedValue = value ?? 'N/A';
```

Comment:
can you create the utils for this?

Reply:
could please point this out once more in the code?